src/core/manager.c

```diff
@@ -556,6 +558,42 @@ static int manager_default_environment(Manager *m) {
         return 0;
 }
 
+static int manager_setup_prefix(Manager *m) {
+        ExecDirectoryType i;
+        int r;
+        struct {
+                uint64_t type;
+                const char *suffix;
+        } paths_system[_EXEC_DIRECTORY_MAX] = {
+                [EXEC_DIRECTORY_RUNTIME] = { SD_PATH_SYSTEM_RUNTIME, NULL },
+                [EXEC_DIRECTORY_STATE] = { SD_PATH_SYSTEM_STATE_PRIVATE, NULL },
+                [EXEC_DIRECTORY_CACHE] = { SD_PATH_SYSTEM_STATE_CACHE, NULL },
+                [EXEC_DIRECTORY_LOGS] = { SD_PATH_SYSTEM_STATE_LOGS, NULL },
+                [EXEC_DIRECTORY_CONF] = { SD_PATH_SYSTEM_CONFIGURATION, NULL },
+        }, paths_user[_EXEC_DIRECTORY_MAX] = {
+                [EXEC_DIRECTORY_RUNTIME] = { SD_PATH_USER_RUNTIME, NULL },
+                [EXEC_DIRECTORY_STATE] = { SD_PATH_USER_CONFIGURATION, NULL },
+                [EXEC_DIRECTORY_CACHE] = { SD_PATH_SYSTEM_STATE_CACHE, NULL },
+                [EXEC_DIRECTORY_LOGS] = { SD_PATH_SYSTEM_CONFIGURATION, "log" },
+                [EXEC_DIRECTORY_CONF] = { SD_PATH_SYSTEM_CONFIGURATION, NULL },
+        }, *p;
```

Comment:
this looks great, but it it drops the "static const" bit on the actual tables. I think this should remain inside manager_setup_prefix()'s local scope, but be changed to this:

```c
struct table_entry {
        uint64_t type;
        const char *suffix;
};

static const struct table_entry paths_system[_EXEC_DIRECTORY_MAX] = {
…
}

static const struct table_entry paths_user[_EXEC_DIRECTORY_MAX] = {
…
}

const struct table_entry *p;

…
```